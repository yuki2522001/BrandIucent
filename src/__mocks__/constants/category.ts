import { Categories } from "@common-types/category";
import url from "@assets/images/icons/beverages.png";

const CATEGORY: Categories = {
  id: "1234",
  name: "Beverages",
  images: {
    src: url,
    alt: "This is icon beverages",
  },
};

const CATEGORY_MOCKING_LIST: Categories[] = [
  {
    id: "1234",
    name: "Beverages",
    images: {
      src: "@assets/images/icons/beverages.png",
      alt: "This is icon beverages",
    },
  },
  {
    id: "12345",
    name: "Dairy",
    images: {
      src: "@assets/images/icons/dairy.png",
      alt: "This is icon dairy",
    },
  },
  {
    id: "12345",
    name: "Meat",
    images: {
      src: "@assets/images/icons/meat.png",
      alt: "This is icon meat",
    },
  },
  {
    id: "12345",
    name: "Spices & Herbs",
    images: {
      src: "@assets/images/icons/spices-herbs.png",
      alt: "This is icon spices-herbs",
    },
  },
  {
    id: "12345",
    name: "Stationery",
    images: {
      src: "@assets/images/icons/stationery.png",
      alt: "This is icon stationery",
    },
  },
  {
    id: "12345",
    name: "Costmetic",
    images: {
      src: "@assets/images/icons/costmetic.png",
      alt: "This is icon costmetic",
    },
  },
];

export { CATEGORY, CATEGORY_MOCKING_LIST };
