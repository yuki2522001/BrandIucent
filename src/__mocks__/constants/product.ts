import { Product } from "@common-types/product";
import iceCreamSundae from "@assets/images/ice-cream-sundae.png";
import beefSteak from "@assets/images/beef-steak.png";

const PRODUCT_MOCKING: Product = {
  productId: "16466545464646547",
  categoryId: "1651454657522",
  productName: "Ice Cream Sundae",
  productDescription:
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam molestie ultrices donec pulvinar quis est bibendum suspendisse viverra.",
  productQuantity: 5,
  originalPrice: {
    value: 20,
    currency: "$",
  },
  discountPrice: {
    value: 20,
    currency: "$",
  },
  offerQuantity: "3kg",
  offer: false,
  bestSelling: true,
  popular: false,
  images: {
    src: iceCreamSundae,
    alt: "This is ice-cream-sundae images",
  },
};
const PRODUCT_MOCKING_LIST: Product[] = [
  {
    productId: "16466545464646547",
    categoryId: "1651454657522",
    productName: "Ice Cream Sundae",
    productDescription:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam molestie ultrices donec pulvinar quis est bibendum suspendisse viverra.",
    productQuantity: 5,
    originalPrice: {
      value: 20,
      currency: "$",
    },
    discountPrice: {
      value: 20,
      currency: "$",
    },
    offerQuantity: "3kg",
    offer: false,
    bestSelling: true,
    popular: false,
    images: {
      src: iceCreamSundae,
      alt: "This is ice-cream-sundae images",
    },
  }
];

export { PRODUCT_MOCKING, PRODUCT_MOCKING_LIST };
