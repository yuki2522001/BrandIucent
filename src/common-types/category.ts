export interface Categories {
  id: string;
  name: string;
  images: {
    src: string;
    alt: string;
  };
}