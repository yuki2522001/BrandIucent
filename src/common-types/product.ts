export interface Product {
  productId: string;
  categoryId: string;
  productName: string;
  productDescription: string ;
  productQuantity: number;
  originalPrice: {
    value: number;
    currency: string;
  };
  discountPrice: {
    value: number;
    currency: string;
  };
  offerQuantity: string;
  offer: boolean;
  bestSelling: boolean;
  popular: boolean;
  images: {
    src: string;
    alt: string;
  }
}

export interface ProductProps {
  product?: Product;
}