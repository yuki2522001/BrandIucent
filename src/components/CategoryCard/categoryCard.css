.categoryCard {
  width: 223px;
  display: flex;
  flex-direction: column;
}

.categoryCard__image {
  margin: 32px 0 60px;
  text-align: center;
}

.categoryCard__text {
  margin: 0 15px 45px 70px;
  font-weight: 400;
  font-size: 18px;
  line-height: 21px;
  color: var(--dark);
  text-decoration: none;
}