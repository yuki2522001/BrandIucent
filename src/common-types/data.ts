import { Categories } from "./category";
import { Product } from "./product";

export interface DataContextProps {
  products: Product[];
  setProducts: Function;
  searchValue: [] | undefined;
  setSearchValue: Function;
  categories: Categories[];
  setCategories: Function;
}