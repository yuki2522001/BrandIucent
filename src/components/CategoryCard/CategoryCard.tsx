import React, { memo } from "react";
import { Link } from "react-router-dom";
import { Categories } from "@common-types/category";
import "./categoryCard.css";

interface CategoryCardProps {
  category: Categories;
  categoryId: string;
  onToggleCategory: Function;
}

const CategoryCard: React.FC<CategoryCardProps> = memo(
  ({ onToggleCategory, category, categoryId }) => {
    const handleSelect = () => {
      onToggleCategory(categoryId);
    };

    return (
      <div data-testid="category-card" className="categoryCard">
        <figure className="categoryCard__image">
          <img src={category.images.src} alt={category.images.alt} />
        </figure>
        <Link
          className="categoryCard__text"
          to={`/products/category/${category.id}`}
          onClick={handleSelect}
        >
          <p data-testid="category-item">{category.name}</p>
        </Link>
      </div>
    );
  }
);

export default CategoryCard;
