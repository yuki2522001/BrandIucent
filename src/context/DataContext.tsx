import { createContext, useMemo, useState } from "react";
import { Categories } from "@common-types/category";
import { DataContextProps } from "@common-types/data";
import { Product } from "@common-types/product";

const DataContext = createContext<DataContextProps>({} as DataContextProps);
const DataProvider: React.FC<{ children: JSX.Element[] | JSX.Element }> = ({
  children,
}) => {
  const [searchValue, setSearchValue] = useState<[]>([]);
  const [categories, setCategories] = useState<Categories[]>([]);
  const [products, setProducts] = useState<Product[]>([]);

  const value = {
    products,
    setProducts,
    searchValue,
    setSearchValue,
    categories,
    setCategories,
  };

  return <DataContext.Provider value={value}>{children}</DataContext.Provider>;
};

export { DataProvider, DataContext };
