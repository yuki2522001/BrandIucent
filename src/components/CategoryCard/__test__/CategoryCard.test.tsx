import { render } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";
import { DataContextProps } from "@common-types/data";
import { DataContext } from "@context/DataContext";
import { CATEGORY, CATEGORY_MOCKING_LIST } from "@__mocks__/constants/category";
import { PRODUCT_MOCKING_LIST } from "@__mocks__/constants/product";
import "@testing-library/jest-dom";
import CategoryCard from "../CategoryCard";

const contextProductMock: DataContextProps = {
  searchValue: [],
  setSearchValue: jest.fn(),
  categories: CATEGORY_MOCKING_LIST,
  setCategories: jest.fn(),
  products: PRODUCT_MOCKING_LIST,
  setProducts: jest.fn(),
};

describe("Categories component", () => {
  test("should render categories component", () => {
    const history = createMemoryHistory();
    const { getByTestId } = render(
      <DataContext.Provider value={contextProductMock}>
        <Router location={history.location} navigator={history}>
          <CategoryCard category={CATEGORY} categoryId={""} onToggleCategory={() => {}} />
        </Router>
      </DataContext.Provider>
    );
    expect(getByTestId("category-card")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const history = createMemoryHistory();
    const { asFragment } = render(
      <DataContext.Provider value={contextProductMock}>
        <Router location={history.location} navigator={history}>
        <CategoryCard category={CATEGORY} categoryId={""} onToggleCategory={() => {}} />
        </Router>
      </DataContext.Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
