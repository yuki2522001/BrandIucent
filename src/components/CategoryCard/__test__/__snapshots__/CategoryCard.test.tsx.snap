// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="categoryCard"
    data-testid="category-card"
  >
    <figure
      class="categoryCard__image"
    >
      <img
        alt="This is icon beverages"
      />
    </figure>
    <a
      class="categoryCard__text"
      href="/products/category/1234"
    >
      <p
        data-testid="category-item"
      >
        Beverages
      </p>
    </a>
  </div>
</DocumentFragment>
`;
